x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -85,6 +73,29 @@ export function isMappableJob(job: CombinedJob, detectorIndex: number): boolean
   return isMappable;
 }
 
+export function hasValidComposite(buckets: { [key: string]: any }) {
+  if (
+    isPopulatedObject(buckets, ['composite']) &&
+    isPopulatedObject(buckets.composite, ['sources'])
```

Comment:
what do you think about extending `isPopulatedObject` guard to support nested props?
```suggestion
    isPopulatedObject(buckets, ['composite.sources'])
```

Reply:
I think this is a good idea although I think we probably should have a different API for handling that type of nested cases as it is possible for an object to have keys with dots in the name.
`const myObj = {"composite.sources": "value1", "composite": { "source": "value2" }}`